### PR TITLE
Route brace expansion

### PR DESCRIPTION
**Pull request by riga**
This PR provides brace expansion to all places where we handle multiple user defined routes / columns. As a result, it becomes possible to write e.g.

```python
@producer(
   uses={"Jet.{pt,eta,phi}"},
)
def ...
```

in CSPs, or

```python
cfg.x.keep_columns = DotDict.wrap({
    "cf.ReduceEvents": {
        "{Jet,FatJet}.{pt,eta,phi,mass,btagPNet*}",
```

in the config. On our end, this would make longish configurations way easier to understand and maintain.

### CODE REVIEW

File: columnflow/tasks/reduction.py [hunk 1]
```diff
@@ -113,7 +113,15 @@ def run(self):
...
         skip_columns: set[str] = set()
         for c in self.config_inst.x.keep_columns.get(self.task_family, ["*"]):
-            for r in (self.find_keep_columns(c) if isinstance(c, ColumnCollection) else {Route(c)}):
+            # expand collections and braces in strings
+            if isinstance(c, ColumnCollection):
+                routes = self.find_keep_columns(c)
+            elif isinstance(c, str):
+                routes = set(map(Route, law.util.brace_expand(c)))
+            else:
+                routes = {Route(c)}
```
Comment (mafrahm) [inline columnflow/tasks/reduction.py:122]: This code block of determining the routes seems to repeat quite often. Does it make sense to refactor this?

File: columnflow/columnar_util.py [hunk 2]
```diff
@@ -366,6 +366,15 @@ def __eq__(self, other: Route | str | Sequence[str | int | slice | type(Ellipsis
             return self.column == other
         return False
 
+    def __lt__(self, other: Route | str | Sequence[str | int | slice | type(Ellipsis) | list]) -> bool:
```
Comment (mafrahm) [inline columnflow/columnar_util.py:369]: is this used anywhere?

**Review by mafrahm — APPROVED** [review]
LGTM, just two minor inline comments

File: columnflow/columnar_util.py (hunk 2 above)
Comment (riga) [reply columnflow/columnar_util.py:369]: It's needed when sorting sequences of Route objects. We already have `__eq__`, so `__lt__` was the only missing piece.

File: columnflow/tasks/reduction.py (hunk 1 above)
Comment (riga) [reply columnflow/tasks/reduction.py]: Yup, did that in a new method :+1:

**Pull request merged**